tests/console/validate_raid_encrypt_home.pm

```diff
@@ -0,0 +1,31 @@
+# SUSE's openQA tests
+#
+# Copyright 2024 SUSE LLC
+# SPDX-License-Identifier: FSFAP
+
+# Summary: Simple RAID partitioning layout validation for encrypted home
+# Maintainer: QE YaST and Migration (QE Yam) <qe-yam at suse de>
+
+use strict;
+use warnings;
+use base "opensusebasetest";
+use testapi;
+use Mojo::JSON qw(decode_json);
+
+sub run {
+    select_console 'root-console';
+
+    # To avoid the regression of bsc#1224393, need ensure that the crypt /home
+    # partition is under 'md1' via comparing with the partitioning output of md1.
+    my $lsblk_output = decode_json(script_output("lsblk -M -J /dev/md1"));
+    my $dev_name = $lsblk_output->{blockdevices}->[0]->{name};
+    my $dev_children_name = $lsblk_output->{blockdevices}->[0]->{children}->[0]->{name};
+    my $dev_child_mountpoint = $lsblk_output->{blockdevices}->[0]->{children}->[0]->{mountpoints}->[0];
+    unless (($dev_name =~ 'md1') && ($dev_children_name =~ 'cr_home') && ($dev_child_mountpoint =~ 'home')) {
+        die "RAID partitioning layout not as expected";
```

Comment:
you don't need to create customer die message, we should use library `use Test::Assert ':all'` to work with the expected values from test data and compare with the other values we read from the system with the command. I guess we need 3 pieces of information, name of the block device and type and mount point of the children, those 3 compared should work.

Reply:
The TEST_DATA has a wrong for filesystem: crypto_LUKS, updated to xfs, only crypt_method: luks1 could be used to verify encrypted. So I updated code to get the value of crypt_method and set type = encrypt to compare. 

 my $test_data_type = 'encrypt' if $test_data_crypt;

Please have a look whether this can be accepted. Or we can set a user-defined item in TEST_DATA such as type: encrypt to make the compare works. WDYT?